src/Proto.Cluster/Member/MemberList.cs

```diff
@@ -110,18 +114,34 @@ public void UpdateBannedMembers(string[] bannedMembers)
             }
         }
 
+        public string MemberId => _system.Id;
+
         public void UpdateClusterTopology(IReadOnlyCollection<Member> members)
         {
             lock (this)
             {
                 Logger.LogDebug("[MemberList] Updating Cluster Topology");
+                
+                if (_bannedMembers.Contains(_system.Id))
+                {
+                    if (_stopping)
+                    {
+                        return;
+                    }
+
+                    _stopping = true;
+                    Logger.LogCritical("I have been banned, exiting {Id}", MemberId);
+                    _ = _cluster.ShutdownAsync();
```

Comment:
Not sure what would be better here.
We now gossip Banned members across cluster. if a node detects itself has been banned, it will shut down its own cluster.

say the Consul health check fails for a node, all other nodes detect it as no longer part of the cluster, itself also needs to understand this.